distribution.md

```diff
@@ -155,3 +155,73 @@ Where:
     * contain an array of deprecated APIs:
       * should only contain `0.x` or major versions
       * should only contain APIs that correspond to a spec release
+
+### Builder
+
+The following specifies the artifact format for a buildpacks builder.
+
+A Builder is an OCI Image that provides a distributable build environment.
+
+A platform supporting the builder SHOULD allow users to configure the build environment with a provided builder.
+
+#### General Requirements
+
+The Builder image MUST contain an implementation of the [lifecycle](#lifecycle), and [build-time](#build-image) environment and MAY contain [buildpacks](#buildpackage). Platforms SHOULD use builders to ease the build process. 
+
+#### Environment Variables
+
+A Builder MUST be an extension of a Build Image, and MUST retain all the specified environment variables set on the original build image, as specified in the Build Image specifications.
+
+#### Labels
+
+A Builder MUST be an extension of a Build Image, and MUST retain all the specified Labels set on the original build image, as specified in the Build Image specifications.
+
+A Builder image MUST contain an implementation of the [lifecycle](#lifecycle), and MUST retain all the specified Labels set on the original Lifecycle image, as specified in the lifecycle distribution specifications.
+
+A Builder image MAY contain [buildpacks](#buildpackage), and MAY retain all the specified Labels set on the original buildpackage, as specified in the buildpackage specifications.
```

Comment:
Should we say where the buildpacks live on the filesystem?

Reply:
/cc @jromero

Reply:
I believe that is already covered [by this](https://github.com/buildpacks/spec/blob/cbfaca6e52af1c89b32b0350bffedf9f4c1b4175/distribution.md?plain=1#L43-L49):

![image](https://user-images.githubusercontent.com/475559/144277414-39a0e7bf-ead5-4d5a-937e-75e9e369cbd1.png)


Maybe we can add some additional clarification but I think we should be doing so in the "Buildpackage" section and not in this section. Thoughts?

Reply:
This part seems important from the [previously unreleased version of the builder spec](https://github.com/buildpacks/spec/blob/extensions/builder/0.1/extensions/builder.md#filesystem). I would still propose that this be added to the "buildpackages" section.

![image](https://user-images.githubusercontent.com/475559/144280287-2603c416-d8f8-4570-bcef-af4d17a4c56e.png)

Reply:
I've mentioned the location of the buildpack blob in the buildpackage section as in the link you mentioned above, I'll refine the section.

Reply:
@jromero would adding `CNB_APP_DIR` and `CNB_LAYERS_DIR` in buildpackage section make sense? Because essentially it is referring to the filesystem of the Builder image. We can have a sub-section here, under the builder section and this information there. I left out the path of the buildpack blob because that information is already present in the buildpack specs [here](https://github.com/buildpacks/spec/blob/cbfaca6e52af1c89b32b0350bffedf9f4c1b4175/distribution.md?plain=1#L43-L49).

Reply:
@jromero would adding `CNB_APP_DIR` and `CNB_LAYERS_DIR` in buildpackage section make sense? Because essentially it is referring to the filesystem of the Builder image. We can have a sub-section here, under the builder section and this information there. I left out the path of the buildpack blob because that information is already present in the buildpack specs [here](https://github.com/buildpacks/spec/blob/cbfaca6e52af1c89b32b0350bffedf9f4c1b4175/distribution.md?plain=1#L43-L49).

Reply:
I captured too much in my screenshot. I'm sorry. You are right those don't make sense in the Buildpackages section. We will need them in this PR though. 

I believe the [environment variables](https://github.com/buildpacks/spec/blob/extensions/builder/0.1/extensions/builder.md#environment-variables) defined here need to be included. The exception is `CNB_BUILDPACKS_DIR`. That brings about an issue if we want to reuse layers from Buildpackages. I would propose that we remove this environment variable. cc: @buildpacks/core-team 

---

The part that is missing from my screenshot is:

> If the buildpack ID contains a `/`, it MUST be replaced with `_` in the directory name.

That's very critical.

Reply:
I've added that to the buildpackage section.